boa/src/builtins/mod.rs

```diff
@@ -81,7 +82,9 @@ pub fn init(interpreter: &mut Interpreter) {
         let global = interpreter.global();
         match global {
             Value::Object(ref global_object) => {
-                global_object.borrow_mut().insert_field(name, value);
+                global_object
+                    .borrow_mut()
+                    .insert_property(name, value, Attribute::default());
```

Comment:
Hmmm. There is a problem with this, the problem is that we don't set the right propety attributes for the global object/properties an example of this is the property `undefined` is (READONLY, NON_ENUMERABLE, PERMANENT) but `Object` global object is (`WRITABLE`, `NON_ENUMERABLE`, `CONFIGURABLE`). but here we are setting all as `READONLY`, `NON_ENUMERABLE`, `PERMANENT`.

One way of doing this would be to return a `(&'staitc, Value, Attribute)` with the specified attribute, another way is to have `init` functions not return anything and set the property in the `init` functions

Reply:
Thanks for spotting this.